packages/vue-components/src/utils/utils.js

```diff
@@ -159,3 +159,41 @@ export function VueFixer (vue) {
   vue.mixins.unshift(mixin)
   return vue
 }
+
+// Used in the TipBox component to classify the different styles used by bootstrap from the user input.
+// @params (the user input type of the box)
+export function classifyBootstrapStyle(type, theme) {
+  const defaultStyles
+    = ['warning', 'info', 'definition', 'success', 'danger', 'tip', 'important', 'wrong'];
+  const colorStyles
+    = ['primary', 'secondary', 'success', 'danger', 'warning', 'info', 'light', 'dark', 'muted', 'white'];
+  const defaultStyle = defaultStyles.includes(type) ? type : '';
+  const colorStyle = colorStyles.includes(theme) ? theme : '';
+
+  let primaryStyle;
+
+  switch (defaultStyle) {
+  case 'warning':
+    primaryStyle = 'warning';
+    break;
+  case 'info':
+    primaryStyle = 'info';
+    break;
+  case 'definition':
+    primaryStyle = 'primary';
+    break;
+  case 'success':
+  case 'tip':
+    primaryStyle = 'success';
+    break;
+  case 'important':
+  case 'wrong':
+    primaryStyle = 'danger';
+    break;
+  default:
+    primaryStyle = 'default';
+    break;
+  }
+
+  return {baseStyle: primaryStyle, colorStyle: colorStyle, iconStyle: defaultStyle};
```

Comment:
How about returning either `primaryStyle` or `colorStyle` instead of both since only one of them will be used. This can reduce the use of
```.js
if (colorStyle) {
  return `...-${colorStyle}`;
}
return `...-${baseStyle}`;
```
in some of the methods.

Reply:
Ok this does make more sense, will make this change